lib/IDE/Refactoring.cpp

```diff
@@ -3526,6 +3527,178 @@ performChange() {
   return false;
 }
 
+class AddCodableContext {
+
+  /// Declaration context
+  DeclContext *DC;
+
+  /// Start location of declaration context brace
+  SourceLoc StartLoc;
+
+  /// Array of all conformed protocols
+  SmallVector<swift::ProtocolDecl *, 2> Protocols;
+
+  /// Range of internal members in declaration
+  DeclRange Range;
+
+  bool conformsToCodableProtocol() {
+    for (ProtocolDecl *Protocol : Protocols) {
+      if (Protocol->getKnownProtocolKind() == KnownProtocolKind::Encodable ||
+          Protocol->getKnownProtocolKind() == KnownProtocolKind::Decodable) {
+        return true;
+      }
+    }
+    return false;
+  }
```

Comment:
Could this use `ModuleDecl::lookupConformance(Type, ...)` instead of searching the conformances manually? That would also let you dig out the `init(from:)` or `encode(to:)` witness and check them for `isSynthesized` if we only want to provide this refactoring in cases where it's a synthesized conformance.

Reply:
Ah, you're totally right. With this current implementation we'll provide the refactor option when `Codable` is entirely manually implemented (so there's nothing synthesized). 

I realize it's not ideal in this state, but I'd like to fix this in a follow up PR.

Reply:
Looks like it would appear but do nothing in this state. It's a fairly narrow condition anyway (only on a decl conforming to encodable/decodable), so implementing in a follow up sounds reasonable.